Sage/carbon
# feat(select): add selectionConfirmed property to event emitted on change

fix #6330

### Proposed behaviour


Adds a `CustomSelectChangeEvent` type which defines the shape of the new change event emitted when the input value changes, backwards compatibility is maintained so it is still possible to continue to use React's ChangeEvent. Within the new custom event there is a `selectionConfirmed` property that indicates when a user has pressed `Enter` or clicked on a given `Option`.

Filter out `fill` prop so it isn't set on the underlying DOM element when passed to an `Option`

### Current behaviour


No `CustomSelectChangeEvent` type exists, React's ChangeEvent is what is emitted when input value is updated. Therefore there is currently no way for implementation teams to know when a user has made a selection via keyboard unless the input is blurred when using `Filterable` and `Simple` select components.

When `fill` prop is passed on `Option` it is set on the underlying DOM element.

### Checklist




#### QA


### Additional context


**There wasn't really much need to add the new `selectionConfirmed` event property to `MultiSelect` as it doesn't update the input value as the user keyboard navigates. However, I decided add it anyway for the sake of consistency but I haven't added any public documentation for it.**

### Testing instructions


See new `selection-confirmed-story` added to storybook for Filterable and Simple Select

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
